clarity/contracts/arkadiko-auction-engine-v1-1.clar

```diff
@@ -244,14 +244,14 @@
 )
 
 ;; @desc calculate the discounted auction price on the (dollarcent) price of the collateral
-;; @param price-in-cents; the current on-chain price in cents
+;; @param price; the current on-chain price
 ;; @param auction-id; the ID of the auction in which the collateral will be sold
-(define-read-only (discounted-auction-price (price-in-cents uint) (auction-id uint))
+(define-read-only (discounted-auction-price (price uint) (auction-id uint))
   (let (
     (auction (get-auction-by-id auction-id))
-    (discount (* price-in-cents (get discount auction)))
+    (discount (* price (get discount auction)))
   )
-    (ok (/ (- (* u100 price-in-cents) discount) u100))
+    (ok (/ (- (* u100 price) discount) u1000000))
```

Comment:
Q: we divide by one million here (6 decimals)

what if we handle bitcoin, which has 100M denomination (1 sat = 8 decimals)?

Reply:
The oracle now has 6 decimals to save a price. So BTC price in $ will have 6 decimals, and does not depend on the amount of decimals of the token itself.

Reply:
mm ok.. so which price do we push to the oracle if bitcoin is $50,000 dollar?

Reply:
In that case we set $50.000 with 6 decimals so 50000000000.

( Previously you had the price in cents, so 2 decimals )

Reply:
OK got it! that makes sense